From 8cc35098ccca830f07c6c8cc91173e5d2e09933d Mon Sep 17 00:00:00 2001
From: Steve Lhomme <robux4@ycbcr.xyz>
Date: Tue, 6 Jun 2023 10:04:28 +0200
Subject: [PATCH 4/4] Disable cache for in Universal Windows Platform builds

Calling SHGetFolderPathA is not allowed. It's possible to get the cache
folder with IApplicationData2
https://learn.microsoft.com/en-us/uwp/api/windows.storage.applicationdata
---
 src/libdvdcss.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/libdvdcss.c b/src/libdvdcss.c
index 78067a1..7d65470 100644
--- a/src/libdvdcss.c
+++ b/src/libdvdcss.c
@@ -214,6 +214,7 @@ static int set_cache_directory( dvdcss_t dvdcss )
     if( psz_cache == NULL || psz_cache[0] == '\0' )
     {
 #ifdef _WIN32
+#if defined(WINAPI_FAMILY_PARTITION) && WINAPI_FAMILY_PARTITION (WINAPI_PARTITION_DESKTOP)
         char psz_home[PATH_MAX];
 
         /* Cache our keys in
@@ -225,6 +226,7 @@ static int set_cache_directory( dvdcss_t dvdcss )
             dvdcss->psz_cachefile[PATH_MAX - 1] = '\0';
             psz_cache = dvdcss->psz_cachefile;
         }
+#endif
 #else
 #ifdef __ANDROID__
         /* $HOME is not writable on __ANDROID__ so we have to create a custom
-- 
2.37.3.windows.1

